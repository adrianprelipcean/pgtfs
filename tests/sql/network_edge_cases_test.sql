-- SPDX-FileCopyrightText: Â© 2024 Adrian C. Prelipcean
--
-- SPDX-License-Identifier: EUPL-1.2

create extension if not exists pgtfs;

-- wrong trip id type
select
    stop_id,
    stop_sequence,
    arrival_time,
    trip_id
from pgtfs_csa(
    '1', '2', 1712707200.0,
    $$
select 
    trip_id::int, 
    stop_id, 
    arrival_time::double precision, 
    departure_time::double precision, 
    stop_sequence::int from (
    values
     ('1753','1',NULL,1712757660,1),
     ('1753','2',1712759880,1712759940,2),
     ('1753','3',1712762040,1712762100,3),
     ('1753','4',1712762100,NULL,4)
    )
    as data (trip_id, stop_id, arrival_time, departure_time, stop_sequence);
$$
);

-- wrong stop id type
select
    stop_id,
    stop_sequence,
    arrival_time,
    trip_id
from pgtfs_csa(
    '1', '2', 1712707200.0,
    $$
select 
    trip_id, 
    stop_id::int, 
    arrival_time::double precision, 
    departure_time::double precision, 
    stop_sequence::int from (
    values
     ('1753','1',NULL,1712757660,1),
     ('1753','2',1712759880,1712759940,2),
     ('1753','3',1712762040,1712762100,3),
     ('1753','4',1712762100,NULL,4)
    )
    as data (trip_id, stop_id, arrival_time, departure_time, stop_sequence);
$$
);

-- wrong arrival time type
select
    stop_id,
    stop_sequence,
    arrival_time,
    trip_id
from pgtfs_csa(
    '1', '2', 1712707200.0,
    $$
select 
    trip_id, 
    stop_id, 
    arrival_time::numeric, 
    departure_time::double precision, 
    stop_sequence::int from (
    values
     ('1753','1',NULL,1712757660,1),
     ('1753','2',1712759880,1712759940,2),
     ('1753','3',1712762040,1712762100,3),
     ('1753','4',1712762100,NULL,4)
    )
    as data (trip_id, stop_id, arrival_time, departure_time, stop_sequence);
$$
);

-- wrong departure time type
select
    stop_id,
    stop_sequence,
    arrival_time,
    trip_id
from pgtfs_csa(
    '1', '2', 1712707200.0,
    $$
select 
    trip_id, 
    stop_id, 
    arrival_time::double precision, 
    departure_time::numeric, 
    stop_sequence::int from (
    values
     ('1753','1',NULL,1712757660,1),
     ('1753','2',1712759880,1712759940,2),
     ('1753','3',1712762040,1712762100,3),
     ('1753','4',1712762100,NULL,4)
    )
    as data (trip_id, stop_id, arrival_time, departure_time, stop_sequence);
$$
);

-- wrong stop sequence type
select
    stop_id,
    stop_sequence,
    arrival_time,
    trip_id
from pgtfs_csa(
    '1', '2', 1712707200.0,
    $$
select 
    trip_id, 
    stop_id, 
    arrival_time::double precision, 
    departure_time::double precision, 
    stop_sequence::text from (
    values
     ('1753','1',NULL,1712757660,1),
     ('1753','2',1712759880,1712759940,2),
     ('1753','3',1712762040,1712762100,3),
     ('1753','4',1712762100,NULL,4)
    )
    as data (trip_id, stop_id, arrival_time, departure_time, stop_sequence);
$$
);
