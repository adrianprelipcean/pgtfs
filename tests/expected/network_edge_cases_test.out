-- SPDX-FileCopyrightText: Â© 2024 Adrian C. Prelipcean
--
-- SPDX-License-Identifier: EUPL-1.2
create extension if not exists pgtfs;
NOTICE:  extension "pgtfs" already exists, skipping
-- wrong trip id type
select
    stop_id,
    stop_sequence,
    arrival_time,
    trip_id
from pgtfs_csa(
    '1', '2', 1712707200.0,
    $$
select 
    trip_id::int, 
    stop_id, 
    arrival_time::double precision, 
    departure_time::double precision, 
    stop_sequence::int from (
    values
     ('1753','1',NULL,1712757660,1),
     ('1753','2',1712759880,1712759940,2),
     ('1753','3',1712762040,1712762100,3),
     ('1753','4',1712762100,NULL,4)
    )
    as data (trip_id, stop_id, arrival_time, departure_time, stop_sequence);
$$
);
ERROR:  Expected text for trip ID, got integer
-- wrong stop id type
select
    stop_id,
    stop_sequence,
    arrival_time,
    trip_id
from pgtfs_csa(
    '1', '2', 1712707200.0,
    $$
select 
    trip_id, 
    stop_id::int, 
    arrival_time::double precision, 
    departure_time::double precision, 
    stop_sequence::int from (
    values
     ('1753','1',NULL,1712757660,1),
     ('1753','2',1712759880,1712759940,2),
     ('1753','3',1712762040,1712762100,3),
     ('1753','4',1712762100,NULL,4)
    )
    as data (trip_id, stop_id, arrival_time, departure_time, stop_sequence);
$$
);
ERROR:  Expected text for stop ID, got integer
-- wrong arrival time type
select
    stop_id,
    stop_sequence,
    arrival_time,
    trip_id
from pgtfs_csa(
    '1', '2', 1712707200.0,
    $$
select 
    trip_id, 
    stop_id, 
    arrival_time::numeric, 
    departure_time::double precision, 
    stop_sequence::int from (
    values
     ('1753','1',NULL,1712757660,1),
     ('1753','2',1712759880,1712759940,2),
     ('1753','3',1712762040,1712762100,3),
     ('1753','4',1712762100,NULL,4)
    )
    as data (trip_id, stop_id, arrival_time, departure_time, stop_sequence);
$$
);
ERROR:  Expected float8 for arrival time, got numeric
-- wrong departure time type
select
    stop_id,
    stop_sequence,
    arrival_time,
    trip_id
from pgtfs_csa(
    '1', '2', 1712707200.0,
    $$
select 
    trip_id, 
    stop_id, 
    arrival_time::double precision, 
    departure_time::numeric, 
    stop_sequence::int from (
    values
     ('1753','1',NULL,1712757660,1),
     ('1753','2',1712759880,1712759940,2),
     ('1753','3',1712762040,1712762100,3),
     ('1753','4',1712762100,NULL,4)
    )
    as data (trip_id, stop_id, arrival_time, departure_time, stop_sequence);
$$
);
ERROR:  Expected float8 for departure time, got numeric
-- wrong stop sequence type
select
    stop_id,
    stop_sequence,
    arrival_time,
    trip_id
from pgtfs_csa(
    '1', '2', 1712707200.0,
    $$
select 
    trip_id, 
    stop_id, 
    arrival_time::double precision, 
    departure_time::double precision, 
    stop_sequence::text from (
    values
     ('1753','1',NULL,1712757660,1),
     ('1753','2',1712759880,1712759940,2),
     ('1753','3',1712762040,1712762100,3),
     ('1753','4',1712762100,NULL,4)
    )
    as data (trip_id, stop_id, arrival_time, departure_time, stop_sequence);
$$
);
ERROR:  Expected integer for stop sequence, got text
